scene/3d/baked_lightmap.cpp

```diff
@@ -727,51 +1190,125 @@ Ref<BakedLightmapData> BakedLightmap::get_light_data() const {
 	return light_data;
 }
 
-void BakedLightmap::_debug_bake() {
-	bake(get_parent(), true);
+void BakedLightmap::set_capture_propagation(float p_propagation) {
+	capture_propagation = p_propagation;
+}
+
+float BakedLightmap::get_capture_propagation() const {
+
+	return capture_propagation;
 }
 
-void BakedLightmap::set_propagation(float p_propagation) {
-	propagation = p_propagation;
+void BakedLightmap::set_capture_quality(BakeQuality p_quality) {
+	capture_quality = p_quality;
 }
 
-float BakedLightmap::get_propagation() const {
+BakedLightmap::BakeQuality BakedLightmap::get_capture_quality() const {
+	return capture_quality;
+}
 
-	return propagation;
+void BakedLightmap::set_generate_atlas(bool p_enabled) {
+	generate_atlas = p_enabled;
 }
 
-void BakedLightmap::set_energy(float p_energy) {
-	energy = p_energy;
+bool BakedLightmap::is_generate_atlas_enabled() const {
+	return generate_atlas;
 }
 
-float BakedLightmap::get_energy() const {
+void BakedLightmap::set_max_atlas_size(int p_size) {
+	ERR_FAIL_COND(p_size < 2048);
+	max_atlas_size = p_size;
+}
 
-	return energy;
+int BakedLightmap::get_max_atlas_size() const {
+	return max_atlas_size;
 }
 
 void BakedLightmap::set_bake_quality(BakeQuality p_quality) {
 	bake_quality = p_quality;
+	_change_notify();
 }
 
 BakedLightmap::BakeQuality BakedLightmap::get_bake_quality() const {
 	return bake_quality;
 }
 
-void BakedLightmap::set_bake_mode(BakeMode p_mode) {
-	bake_mode = p_mode;
-}
-
-BakedLightmap::BakeMode BakedLightmap::get_bake_mode() const {
-	return bake_mode;
-}
-
+#ifndef DISABLE_DEPRECATED
```

Comment:
Since we're breaking compat already on the rest of the API, is this worth keeping as deprecated?

Reply:
I think it makes sense to keep it, in case someone opens a project in 3.2.4 and then goes back to 3.2.3 or older. I tested this already and thing don't break too much, you get a broken dependency warning because the resource format can't be loaded, but re-baking lightmaps fixes everything.

It's a very small amount of code anyways, so I'd say we keep it.